# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1298443334 18000
#      Wed Feb 23 01:42:14 2011 -0500
# Node ID c40528d06c9285e4f0b35e8a13a34d6e03b16068
# Parent  454bd50a530848298afdd17eca1f45ef7cd521d1
mex.cc (mxArray_base::dup): return retval

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2011-02-23  John W. Eaton  <jwe@octave.org>
+
+	* mex.cc (mxArray_base::dup): Return retval.
+
 2011-02-22  Konstantinos Poulios  <logari81@googlemail.com>
 
 	* graphics.h.in (class axes::properties): New hidden property
 	looseinset.
 	(axes::properties::update_looseinset): New function.
 	* graphics.cc (axes::properties::init): Initialize looseinset.
 	(axes::properties::sync_positions): Take looseinset into account
 	instead of default_axes_position.
diff --git a/src/mex.cc b/src/mex.cc
--- a/src/mex.cc
+++ b/src/mex.cc
@@ -328,16 +328,18 @@ public:
       id (mxUNKNOWN_CLASS), class_name (0), ndims (-1), dims (0) { }
 
   mxArray *dup (void) const
   {
     mxArray *retval = val.as_mxArray ();
 
     if (! retval)
       retval = new mxArray_octave_value (*this);
+
+    return retval;
   }
 
   ~mxArray_octave_value (void)
   {
     mxFree (class_name);
     mxFree (dims);
   }
 
